### ISSUE TITLE
no unnecessary type annotation

### ISSUE THREAD

**Issue by artola** [helixbass/eslint-plugin-ad-hok#52]
For the following code, 

```ts
addHandlers({
    onChange: ({onChange}): React.ChangeEventHandler<HTMLInputElement> => (event) => {
      onChange(event.target.value);
    },
```

it is shown the following lint error:

```
error  Parameter should be annotated to avoid implicit any-typing  ad-hok/annotate-handler-param-types
```

As workaround I am forced to disable the rule or write the following:

```ts
addHandlers({
    onChange: ({onChange}): React.ChangeEventHandler<HTMLInputElement> => (
      event: React.ChangeEvent<HTMLInputElement>,
    ) => {
      onChange(event.target.value);
    },
```

In my opinion, it is incorrect (or at least  redundancy), due the already existing annotation of the "outer function" that it is a much better approach than just type the individual params.

Could be possible to enhance this detection?
https://github.com/helixbass/eslint-plugin-ad-hok/blob/bdfc2bac7759b2fa3ad8bfecbefa956de8fa2dc7/src/rules/annotate-handler-param-types.coffee#L28

See for example, this existing eslint rule:

https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unnecessary-type-assertion.md

https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/src/rules/no-unnecessary-type-assertion.ts

**Comment by helixbass** [comment]
@artola yes you're right, it doesn't make sense for the rule to flag a handler with an explicit return type on the "outer" handler

And in fact, I don't think I've seen this pattern before but it is quite nice ✨ 

I updated the rule to not flag this case, and released version `0.3.0-beta.5` including this change

cc @peterstuart

**Comment by artola** [comment]
@helixbass Fantastic. Thanks!
